### DISCUSSION TITLE
Hi Xanthic, is there any way to search A or B or (X and Y) in one string?

### DISCUSSION THREAD

**Discussion by lanvuive332** [xanthics/poe_gen_gwennen#11] [Q&A]
for example, i want to search "strenght" and "dexterity" and "8% maximum life"
the number need to be exact
i tried "dexterity|strenght|"8%""um li"" but it didnt work 
thank you

**Comment by xanthics** [comment]
For 8% increased maximum life try `"(?=.*8%)(?=.*life)""str""dex"` and if you want str OR dex `"(?=.*8%)(?=.*life)""str|dex"`  (str or dex might need more characters, I haven't checked for ngram clashing)

From limited testing it seems that each "" group will only match 1 mod on an item.

--------------------------

The search (which is limited to 50 characters, 48 + 2 double quotes) implements most regex rules.  primer here https://medium.com/factory-mind/regex-tutorial-a-simple-cheatsheet-by-examples-649dc1c3f285  Also note that all characters are matched as lower case.

and an example of bracket based or
![image](https://user-images.githubusercontent.com/3137623/147421029-e081ec44-c852-47d5-b317-b07d9ee6e3ca.png)

example of "must have a word starting with c"
![image](https://user-images.githubusercontent.com/3137623/147421039-7cc41dbe-eb81-4617-ad65-ed775298a4ee.png)

"and" operator from https://stackoverflow.com/questions/3041320/regex-and-operator
![image](https://user-images.githubusercontent.com/3137623/147421156-1c8fbb86-7b11-4600-ba20-a20b21b621cd.png)

"and + or"
![image](https://user-images.githubusercontent.com/3137623/147421321-95a24365-21d3-4fcf-b753-0f1ec68c0f25.png)

an example of a search that would work if all the mods were in the same string
![image](https://user-images.githubusercontent.com/3137623/147421448-7172fcd0-191b-4652-845d-80e6d4edf65a.png)

Example of a complex search that works
![image](https://user-images.githubusercontent.com/3137623/147421715-8c3fa253-8911-46b5-a28c-645818e66acd.png)

**Reply by lanvuive332** [reply]
thank you so much for your very detailed answer, have a good new year sir
